.buildkite/utils.jl

```diff
@@ -0,0 +1,88 @@
+import YAML
+
+if !@isdefined(DEBUG)
+    const DEBUG = true
+end
+
+function upload_pipeline(definition)
+    @info "Uploading pipeline..."
+    if DEBUG
+        YAML.write(stderr, definition)
+    else
+        open(`buildkite-agent pipeline upload`, stdout, write=true) do io
+            YAML.write(io, definition)
+        end
+    end
+end
+
+function annotate(annotation; context="default", style="info", append=true)
+    @assert style in ("success", "info", "warning", "error")
+    @info "Uploading annotation..."
+    if DEBUG
+        write(stderr, annotation, '\n')
+    else
+        append = append ? `--append` : ``
+        cmd = `buildkite-agent annotate --style $(style) --context $(context) $(append)`
+        open(cmd, stdout, write=true) do io
+            write(io, annotation)
+        end
+    end
+end
+
+agent() = Dict(
+    :queue => "juliaecosystem",
+    :arch => "x86_64",
+    :os => "linux",
+    :sandbox_capable => "true"
+)
+
+plugins() = [
+    "JuliaCI/julia#v1" => Dict(
+        "persist_depot_dirs" => "packages,artifacts,compiled",
+        "version" => "1.7"
+    )
+]
+
+wait_step() = Dict(:wait => "~")
+group_step(name, steps) = Dict(:group => name, :steps => steps)
+
+function jll_init_step(NAME, PROJECT, BB_HASH, PROJ_HASH)
+    Dict(
+        :label => "jll_init -- $NAME",
+        :agents => agent(),
+        :plugins => plugins(),
+        :timeout_in_minutes => 60,
+        :concurrency => 1,
+        :concurrency_group => "yggdrasil/jll_init",
```

Comment:
This means that even if there are multiple PRs open at the same time, each with their own `init` step, the `init` steps from different PRs will not be allowed to run at the same time. Is this the intended behavior here?

Reply:
The init step shouldn't run at all in pull requests

Reply:
Yeah since builds to `LLVM_full_jll@12` and `LLVM_full_jll@13` will conflict. 
And while we could do some filter magic and remove the version number, a global concurrency limit shouldn't hurt either.

Reply:
Can we add a Buildkite conditional to the `init` step of the form:

```
if: build.branch == "master"
```

Reply:
Ah, I guess you can just check that in your `generator.jl` script.

Reply:
Yeah I decided to just parameterize it in the generator